### DISCUSSION TITLE
Talk Proposal: Combining the Best of Pharaoh, Caesar, and 0 AD for a New City-Building Experience

### DISCUSSION THREAD

**Discussion by Pungolo** [dalerank/Akhenaten#159] [Ideas]
# Talk Proposal: Combining the Best of *Pharaoh*, *Caesar*, and *0 AD* for a New City-Building Experience

Hi everyone, I hope I’m not out of place here, but I’d like to share an idea that I’ve been thinking about for some time. For the past few years, I’ve been playing *Akhenaten*, *Caesaria*, and *0 AD*, and I’ve come to appreciate the strengths of each of these games. But more importantly, I’ve also noticed that there are features in each one that could complement each other in interesting ways. I’d love to propose a concept where we could combine elements from all three games to create a new city-building experience that integrates the best aspects of each.

I should clarify that this is just an idea at the moment—there aren’t any repositories or official plans to implement this yet—but I think it’s a direction worth considering, especially given the dedication of the communities behind each of these titles.

## The Core Idea

At the heart of this idea is using *0 AD* as the base for the game engine. *0 AD* already provides a solid foundation with its extensible graphics and mechanics, such as resource gathering, building construction, unit recruitment, research, and expansion. These are core aspects of any good city-building game, and *0 AD* does them well.

However, I believe we could enhance it by integrating some key elements from *Caesar* and *Pharaoh*, particularly their more complex city-building mechanics. For example, we could add new parameters like hunger, employment, happiness, diseases, and the arrangement of buildings into city zones and work zones. These dynamics would make the player’s role as a ruler even more engaging, as they would have to balance not only economic growth but also the well-being of their citizens.

## Borrowing Elements from *Caesar* and *Pharaoh*

The idea would be to incorporate the economic and administrative systems found in *Caesar* and *Pharaoh*—two games that really shine when it comes to the intricate dynamics of managing a city.

In *Caesar*, you have key factors like Prosperity, Culture, Benevolence, and Peace. Each of these factors could be used to create a system of checks and balances in the new game. For example, if a city becomes too poor, the people might riot; if the culture is neglected, happiness might decrease; and if peace is threatened, the city could suffer from external invasions or internal strife.

Similarly, *Pharaoh* introduced the idea of complex infrastructure, where certain buildings needed to be placed in specific ways to meet the needs of your citizens. In the proposed game, we could extend that idea by incorporating agricultural systems, like building farms on the fertile banks of the Nile or requiring fields of laborers to support the agricultural economy.

But the innovations don’t stop there. *Pharaoh* also required players to build complex monuments, such as obelisks, pyramids, and the Sphinx. These could be important quests or objectives in the new game, pushing players to work toward large-scale goals while managing the day-to-day needs of their city.

## Why This Could Work

What’s great about *0 AD* is that its engine and graphics are flexible and extensible. By using a mod or integrating these new features directly into the core code, we could expand *0 AD* into a more complex, multi-faceted city-building game without losing the fast-paced, real-time strategy mechanics that make it enjoyable.

Since *0 AD* already has a system that is quite similar to that of *Pharaoh*, adding some of the city-building mechanics from *Caesar* and *Pharaoh* could feel like a natural progression rather than a radical departure. You could still build your military and fight battles, but now you’d also need to balance the management of a thriving, prosperous city.

## How I’m Approaching This

As a young developer, I’m still learning the ropes, but I’m very interested in exploring how to make this vision a reality. I’ve started reading through the source code for *0 AD*, *Caesaria*, and *Pharaoh*, and I’ve even opened a post on the *0 AD* forums to gather feedback and suggestions from the community.

Here are some of the resources I’ve been exploring, and I’d love for anyone interested to check them out and possibly collaborate:

https://github.com/azayrahmad/0AD-City-Builder

* https://wildfiregames.com/forum/topic/25273-0ad-economy-simulation-mod/

* https://wildfiregames.com/forum/topic/26701-city-building-mod-a23-a24/

* https://github.com/eserlxl/Arch-AI ( ca be funny for update petra ai)

* https://github.com/PhiGei2000/CityBuilderGame

https://github.com/bvschaik/julius

https://github.com/Keriew/augustus 

https://github.com/dalerank/caesaria-game

There’s still a lot to figure out, but I think this concept has a lot of potential. I really respect the work the communities of these games have put in, and I don’t want to distort the core ideas of any of these titles. Instead, I hope this can be seen as a proposal for improvement and collaboration, a way to expand upon what already exists while staying true to the essence of each game.

I’m not looking to step on anyone’s toes or reinvent the wheel, but I think that combining these elements could create something really exciting for the city-building genre. If you’re interested in collaborating or just want to share your thoughts, I’d love to hear from you.

Thanks for your time, and I look forward to hearing your feedback!*

**Comment by dalerank** [comment]
Hi! I think this is an exciting concept with great potential to deepen the city-building genre.  Those games are result of works from many peoples and years. Just make a plan and moving forward, you can do all you described. Start from something simple, just draw map and couple buildings with simple mechanics. 
I will close this comment next week, because it out of scope this project.

**Comment by Pungolo** [comment]
thank you for your reply. Ok sorry in fact six I wrote here.

If anyone is interested please contact me .

Thank you all for your work and sfornzo for the community

**Comment by Vilrockerdefer** [comment]
Hello,

Thank you for your work…

The android build is unfortunately flagged as infected. On virustotal : Trojan-Spy.AndroidOS.Banker

**Comment by Vilrockerdefer** [comment]
And the installed Akhenaten dev doesn't run at least on my Galaxy tab s2 android 7 tablet.

**Comment by dalerank** [comment]
For virus it stranged, guess to build it from sources in that case. I have only github actions pipeline for create it.
@Vilrockerdefer for the second message pls create issue here
